packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -275,23 +275,23 @@ export class SummaryGenerator {
 		// Use record type to prevent unexpected value types
 		let summaryData: SubmitSummaryResult | undefined;
 		try {
+			// Need to record before we record new attempt
+			const lastAttemptRefSeqNum = this.heuristicData.lastAttempt.refSequenceNumber;
+
 			summaryData = await this.submitSummaryCallback({
 				fullTree,
 				refreshLatestAck,
 				summaryLogger: logger,
 				cancellationToken,
 			});
 
-			this.heuristicData.recordAttempt(summaryData.referenceSequenceNumber);
-
 			// Cumulatively add telemetry properties based on how far generateSummary went.
 			const referenceSequenceNumber = summaryData.referenceSequenceNumber;
 			summarizeTelemetryProps = {
 				...summarizeTelemetryProps,
 				referenceSequenceNumber,
 				minimumSequenceNumber: summaryData.minimumSequenceNumber,
-				opsSinceLastAttempt:
-					referenceSequenceNumber - this.heuristicData.lastAttempt.refSequenceNumber,
+				opsSinceLastAttempt: referenceSequenceNumber - lastAttemptRefSeqNum,
```

Comment:
>opsSinceLastAttempt: referenceSequenceNumber - lastAttemptRefSeqNum, [](http://example.com/codeflow?start=4&length=68)

I thought this data was being logged correctly (Summarize_generate, Summarize_op, etc.)